### PR TITLE
Implementation of PPCWMAX

### PR DESCRIPTION
**Pull request by svenn-t**
This implements the PPCWMAX keyword to limit the maximum capillary pressure scaling when using SWATINIT. The re-calculation of initial water saturation is in the companion opm-simulator PR: https://github.com/OPM/opm-simulators/pull/4707

### CODE REVIEW

File: src/opm/material/fluidmatrixinteractions/EclMaterialLawManager.cpp [hunk 1]
```diff
@@ -33,6 +33,7 @@
 #include <opm/material/fluidstates/SimpleModularFluidState.hpp>
 
 #include <algorithm>
+#include <iostream>
```
Comment (totto82) [inline src/opm/material/fluidmatrixinteractions/EclMaterialLawManager.cpp:37]: Left over from debuggin?

File: src/opm/input/eclipse/EclipseState/Tables/Ppcwmax.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/*
+  Copyright 2023 NORCE
+
```
Comment (totto82) [inline src/opm/input/eclipse/EclipseState/Tables/Ppcwmax.cpp:3]: Equinor

File: opm/input/eclipse/EclipseState/Tables/Ppcwmax.hpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
+/*
+  Copyright (C) 2023 NORCE
+
```
Comment (totto82) [inline opm/input/eclipse/EclipseState/Tables/Ppcwmax.hpp:3]: Equinor

**Review by totto82 — APPROVED** [review]
Thanks. Looks good. Added some minor nitpicks.

File: src/opm/material/fluidmatrixinteractions/EclMaterialLawManager.cpp (hunk 1 above)
Comment (svenn-t) [reply src/opm/material/fluidmatrixinteractions/EclMaterialLawManager.cpp]: Yes, sorry about that : p

File: src/opm/input/eclipse/EclipseState/Tables/Ppcwmax.cpp (hunk 2 above)
Comment (svenn-t) [reply src/opm/input/eclipse/EclipseState/Tables/Ppcwmax.cpp:3]: Changed

File: opm/input/eclipse/EclipseState/Tables/Ppcwmax.hpp (hunk 3 above)
Comment (svenn-t) [reply opm/input/eclipse/EclipseState/Tables/Ppcwmax.hpp:3]: Changed

File: src/opm/input/eclipse/EclipseState/Tables/Ppcwmax.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+std::vector< PpcwmaxRecord >::const_iterator Ppcwmax::end() const {
+    return this->data.end();
+}
+
+Ppcwmax Ppcwmax::serializationTestObject() {
+    Ppcwmax ppcwmax;
+    return ppcwmax;
+}
```
Comment (bska) [inline src/opm/input/eclipse/EclipseState/Tables/Ppcwmax.cpp:71]: For future reference, the intention of `serializationTestObject()` is to return an object that is **not** just default-constructed.  Default-constructed objects don't really exercise the serialisation code.

File: src/opm/material/fluidmatrixinteractions/EclMaterialLawManager.cpp [hunk 5]
```diff
@@ -81,6 +81,22 @@ initFromState(const EclipseState& eclState)
                 stoneEtas_.push_back(table.eta);
             }
         }
+        
+        const auto& ppcwmaxTables = tables.getPpcwmax();
+        this->enablePpcwmax_ = !ppcwmaxTables.empty() ? true : false;
```
Comment (bska) [inline src/opm/material/fluidmatrixinteractions/EclMaterialLawManager.cpp:86]: It's slightly more idiomatic to write this as
```C++
this->enablePpcxwmax_ = !ppcwmaxTables.empty();
```
The `.empty()` function already returns a `bool` so you don't need the explicit `true` and `false` values here.

**Review by bska — COMMENTED** [review]
I did not look at the implementation of the feature, merely at the C++ part.  Except for a couple of minor issues this looks good to me.

File: src/opm/input/eclipse/EclipseState/Tables/Ppcwmax.cpp (hunk 4 above)
Comment (svenn-t) [reply src/opm/input/eclipse/EclipseState/Tables/Ppcwmax.cpp]: Thanks for pointing this out. I agree, so I changed the code to test for non-default records.

File: src/opm/material/fluidmatrixinteractions/EclMaterialLawManager.cpp (hunk 5 above)
Comment (svenn-t) [reply src/opm/material/fluidmatrixinteractions/EclMaterialLawManager.cpp]: Thanks, I changed this.

**Comment by bska** [comment]
jenkins build this opm-simulators=4707 please

**Comment by bska** [comment]
jenkins build this opm-simulators=4707 update_data please

**Comment by bska** [comment]
jenkins build this opm-simulators=4707 opm-tests=994 please

**Comment by bska** [comment]
Thanks a lot for the updates.  The new reference solution has been installed on the CI system so I'll merge this into the master branch now.

**Pull request merged**